charts/grafana/templates/deployment.yaml

```diff
@@ -14,7 +14,7 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
-  {{- if (not .Values.autoscaling.enabled) }}
+  {{- if and (not .Values.autoscaling.enabled) (hasKey .Values "replicas") }}
```

Comment:
To my best knowledge, this does not work, if grafana is embedded as sub-chart

Reply:
This issue seems to be resolved in the recent versions